subiquity/cloudinit.py

```diff
@@ -67,6 +85,56 @@ def read_legacy_status(stream):
     return None
 
 
+def read_recoverable_errors(
+    stream: bytes | str,
+    level: Optional[str] = None,
+) -> list[str]:
+    """Retrieve recoverable errors from cloud-init status."""
+
+    try:
+        status: dict[str, Any] = json.loads(stream)
+    except json.JSONDecodeError:
+        return []
```

Comment:
I think a warning would be good here. Not sure if we should try to display the contents of `stream` in the warning message though...

Reply:
I think that's a good idea. It has to expect the output from `cloud-init status --format=json` so I think something like "Failed to decode cloud-init status output" is informative enough.